### ISSUE TITLE
Can't unset cheat hotkey

### ISSUE THREAD

**Issue by cd4053** [mamedev/mame#10387]
mame: `0.248 (mame0248-72-gcd6d4cbb780)`
OS: `Debian 11`
command: `./mame -window -norc -rompath roms -plugins -plugin cheat -sound none sf2`

1. Press Tab > Plugin Option > Cheat > Set hotkeys (choose and set something) > Done > Quit mame
2. Press Tab > Plugin Option > Cheat > Set hotkeys > Press Delete to unset the hotkey > Done > Return to main screen > Quit mame
3. Start mame again, the hotkey still works after deletion.

Checking the **cheat** folder, the `sf2_hotkeys.json` file still there with previous deleted hotkey intact:

```
[{
    "keys":"KEYCODE_SPACE",
    "desc":"P1 Invincibility"
  }]
```
The only way to unset the hotkey is to manually delete `sf2_hotkeys.json` file.

**Comment by cuavas** [comment]
Oh, this is an interesting bug.  It’s only a problem if you clear all the hotkeys for the system.  If you clear one hotkey while leaving another set, it should behave correctly.

**Comment by cuavas** [comment]
Fixed by bdfdbd8796e8b01cbede23c7e50e812628fda05c

**Comment by cd4053** [comment]
Thanks.
